src/main/java/com/adobe/aem/compgenerator/utils/DialogUtils.java

```diff
@@ -42,33 +44,66 @@
      */
     public static void createDialogXml(final GenerationConfig generationConfig, final String dialogType) {
         String dialogPath = generationConfig.getCompDir() + "/" + dialogType;
-        try {
-            CommonUtils.createFolder(dialogPath);
-
-            Document doc = DocumentBuilderFactory.newInstance().newDocumentBuilder().newDocument();
-            Element rootElement = createDialogRoot(doc, generationConfig, dialogType);
-            List<Property> properties = generationConfig.getOptions().getProperties();
-
-            if (dialogType.equalsIgnoreCase(Constants.DIALOG_TYPE_GLOBAL)) {
-                properties = generationConfig.getOptions().getGlobalProperties();
-            } else if (dialogType.equalsIgnoreCase(Constants.DIALOG_TYPE_SHARED)) {
-                properties = generationConfig.getOptions().getSharedProperties();
-            }
-
-            if (properties != null && properties.size() > 0) {
-                Node currentNode = updateDefaultNodeStructure(doc, rootElement);
-                properties.stream().filter(Objects::nonNull)
-                        .map(property -> createPropertyNode(doc, currentNode, property)).filter(Objects::nonNull)
-                        .forEach(a -> currentNode.appendChild(a));
-            }
-
-            doc.appendChild(rootElement);
-            XMLUtils.transformDomToFile(doc, dialogPath + "/" + Constants.FILENAME_CONTENT_XML);
-        } catch (Exception e) {
+		try {
+			CommonUtils.createFolder(dialogPath);
+
+			Document doc = DocumentBuilderFactory.newInstance().newDocumentBuilder().newDocument();
+			Element rootElement = createDialogRoot(doc, generationConfig, dialogType);
+
+			List<Property> properties = null;
+			
+			if (dialogType.equalsIgnoreCase(Constants.DIALOG_TYPE_DIALOG)) {
+				List<Tab> tabs = generationConfig.getOptions().getTabs();
+				if (null != tabs && tabs.size() > 0) {
+					/* Have not considered global and shared properties when tabs are present. */
```

Comment:
If we make tabs a part of normal dialog rendering with the existing JSON, we shouldnt need any sort of special coding to handle shared/global properties dialogs b/c the logic that generates those dialogs is 100% the same.

Instead of putting "properties" inside of a new "tabs" element in the JSON, I think the simpler and cleaner approach will be to support tabs declared inside of "properties" (which will then allow us to use the same pattern inside of "properties-shared" and "properties-global".

Reply:
I will check on the shared on global properties once again and will revert back on this. Hopefully it should be something like the json provided in the main comment.